infoschema/builder.go

```diff
@@ -67,13 +67,48 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 		return b.applyDropPolicy(diff.SchemaID), nil
 	case model.ActionAlterPlacementPolicy:
 		return b.applyAlterPolicy(m, diff)
+	case model.ActionCreateTables:
```

Comment:
The logic here is very complex.
Please follow `rename tables`

Reply:
And please open a new PR to do the refactoring.

Reply:
I'm happy to do so, but which part do you want me to extract?

Reply:
Refactor the ApplyDiff so that `rename tables` only copy schemaDiff once. Based on that PR, it's easy to handle ApplyDiff for `create tables`